builds/checkin/compatibility-tool-checkin.yaml

```diff
@@ -0,0 +1,25 @@
+trigger: none
+pr:
+  branches:
+    include:
+      - main
+
+jobs:
+  ################################################################################
+  - job: check_run_pipeline
+  ################################################################################
+    displayName: Check pipeline preconditions (changes ARE in platform compatibility script tool)
+    pool:
+      vmImage: "ubuntu-18.04"
+    steps:
+      - checkout: self
+        submodules: false
+        fetchDepth: 3
+      - bash: |
+          git log -m -1 --name-only --first-parent --pretty="" | egrep -i '^(platform-validation/scripts/compatibility)'
+          if [[ $? == 0 ]]; then
+            echo "Detected changes in platform compatibility script tool"
+            sudo apt install shellcheck
+            shellcheck --severity=error ./platform-validation/scripts/compatibility.sh
+          fi
+        displayName: Check changes in platform compatibility script tool and perform shellcheck
```

Comment:
This could be simply "Perform Shell Check" after you remove the check changes code